src/truncate.jl

```diff
@@ -213,6 +213,17 @@ end
 
 ## random number generation
 
+
+"""
+    rand(rng::AbstractRNG, d::Truncated)
+
+Generate a single random sample from a truncated distribution.
+
+The sampling strategy depends on the probability mass of the truncated region (`tp`):
+- If `tp > 0.25`, rejection sampling is used. This is efficient when the truncated region covers a large portion of the original distribution.
+- If `sqrt(eps) < tp <= 0.25`, inverse transform sampling is used. This is more efficient for smaller truncated regions.
+- If `tp` is very small (`<= sqrt(eps)`), a numerically stable version of inverse transform sampling is used which performs calculations in log-space to maintain precision.
+"""
```

Comment:
These are internal implementation details that might change at any point in time and are not relevant to users IMO. There's no need to document the function as its behaviour is the same as `rand(rng, ::Distribution)` for arbitrary distributions.

Reply:
```suggestion
```

Reply:
IMO this example reveals a bug that would be fixed by https://github.com/JuliaStats/Distributions.jl/pull/1986.